integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/allqueries/InsertIntoCarbonTableTestCase.scala

```diff
@@ -67,6 +67,21 @@ class InsertIntoCarbonTableTestCase extends QueryTest with BeforeAndAfterAll {
 
   }
 
+  test("insert from orc-select columns with columns having null values and sort scope as global sort") {
+    sql("drop table if exists TORCSource")
+    sql("drop table if exists TCarbon")
+    sql("create table TORCSource(name string,col array<String>,fee int) STORED AS orc")
+    sql("insert into TORCSource values('karan',null,2)")
+    sql("create table TCarbon(name string, col array<String>,fee int) STORED AS carbondata TBLPROPERTIES ('SORT_COLUMNS'='name','TABLE_BLOCKSIZE'='128','TABLE_BLOCKLET_SIZE'='128','SORT_SCOPE'='global_SORT')")
+    sql("insert overwrite table TCarbon select name,col,fee from TORCSource")
+    val result = sql("show segments for table TCarbon").collect()(0).get(1).toString()
+    if(!"Success".equalsIgnoreCase(result)) {
+      assert(false)
+    }
+    sql("drop table if exists TORCSource")
+    sql("drop table if exists TCarbon")
+  }
+
```

Comment:
please handle and verify 4 scenarios by comparing with ORC.
a) local sort insert with complex type null value data
b) global sort insert with complex type null value data
c) same point a) with `carbon.enable.bad.record.handling.for.insert `as `true`
d) same point b) with `carbon.enable.bad.record.handling.for.insert` as `true`

Reply:
Done